core/services/ocr2/plugins/ocr2keeper/evm21/upkeepstate/orm.go

```diff
@@ -0,0 +1,89 @@
+package upkeepstate
+
+import (
+	"database/sql"
+	"errors"
+	"math/big"
+	"time"
+
+	"github.com/smartcontractkit/sqlx"
+
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+type orm struct {
+	chainID *big.Int
+	q       pg.Q
+}
+
+type PersistedStateRecord struct {
+	UpkeepID            []byte
+	WorkID              string
+	CompletionState     uint8
+	BlockNumber         uint64
+	IneligibilityReason uint8
+	AddedAt             time.Time
+}
+
+// NewORM creates an ORM scoped to chainID.
+func NewORM(chainID *big.Int, db *sqlx.DB, lggr logger.Logger, cfg pg.QConfig) *orm {
+	return &orm{
+		chainID: chainID,
```

Comment:
Might as well convert to `utils.Big` right here
```suggestion
		chainID: utils.NewBig(chainID),
```